### ISSUE TITLE
Server.Utils.ImportCollection of re-zipped collector results with no artifact files crashed the velociraptor process (panic, SIGSEGV)

### ISSUE THREAD

**Issue by chris-coalitionir** [Velocidex/velociraptor#1070]
**Background:**

1. Imported two offline collectors.  First imported with no issues (to rule out any fundamental issues with either the velociraptor install or the general `Server.Utils.ImportCollection` artifact).
2. Second import failed with the error `import_collection: zip: not a valid zip file`.
3. Checked the zip file itself: both 7z and zip could list the file contents, and 7z could extract it into a folder.  At first glance, all looked fine - it contained a `C` folder with all the collected KapeFiles.  It didn't, however, contain the artifact files (e.g., `Windows.KapeFiles.Targets/Uploads.json` and so on).  It's as though it were interrupted before creating them - which may have in fact happened.
4. Before noticing those files were not present, and thinking it was perhaps an issue with the zipping functionality, I re-zipped the results using both 7z and zip and tried to import them with `Server.Utils.ImportCollection`. 
5. Instead of getting the same error (`import_collection: zip: not a valid zip file`), the service immediately crashed.  As in, hard down, immediate network errors in the GUI. Reviewing syslog shows the panic:

```
May 12 13:49:51 hostname velociraptor[461]: [ERROR] 2021-05-12T13:49:51Z Can not create dir: mkdir /opt/vr/clients/C.ea91595f34523baf/collections/F.C2DTPVTF2VAOU/uploads/file/C: not a directory
May 12 13:49:51 hostname velociraptor[461]: panic: runtime error: invalid memory address or nil pointer dereference
May 12 13:49:51 hostname velociraptor[461]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1390a6a]
May 12 13:49:51 hostname velociraptor[461]: goroutine 9226 [running]:
May 12 13:49:51 hostname velociraptor[461]: www.velocidex.com/golang/velociraptor/vql/tools.ImportCollectionFunction.Call.func3(0xc000f0f438, 0xc000f0f518, 0xc0006b1570, 0x1ddca78, 0xc000e33118, 0x1dd18e0, 0xc0028805c0, 0x1dd24e8, 0xc002881dc0)
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/projects/velociraptor/vql/tools/import.go:245 +0x24a
May 12 13:49:51 hostname velociraptor[461]: www.velocidex.com/golang/velociraptor/vql/tools.ImportCollectionFunction.Call(0x1dd18e0, 0xc0028805c0, 0x1df1458, 0xc0003bc770, 0xc002881580, 0x0, 0x0)
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/projects/velociraptor/vql/tools/import.go:260 +0x1954
May 12 13:49:51 hostname velociraptor[461]: www.velocidex.com/golang/vfilter.(*_SymbolRef).callFunction(0xc001454e10, 0x1dd18e0, 0xc0028805c0, 0x1df1458, 0xc0003bc770, 0x1dbad40, 0x2bd6608, 0xc001885720, 0xc00006f8c8)
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20210502142502-4d8770ff3de8/vfilter.go:1990 +0x4bd
May 12 13:49:51 hostname velociraptor[461]: www.velocidex.com/golang/vfilter.(*_SymbolRef).Reduce(0xc001454e10, 0x1dd18e0, 0xc0028805c0, 0x1df1458, 0xc0003bc770, 0x0, 0x0)
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20210502142502-4d8770ff3de8/vfilter.go:1845 +0x7b4
May 12 13:49:51 hostname velociraptor[461]: www.velocidex.com/golang/vfilter.(*_Value).Reduce(0xc0012cc780, 0x1dd18e0, 0xc0028805c0, 0x1df1458, 0xc0003bc770, 0x0, 0x0)
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20210502142502-4d8770ff3de8/vfilter.go:1745 +0x385
May 12 13:49:51 hostname velociraptor[461]: www.velocidex.com/golang/vfilter.(*_MemberExpression).Reduce(0xc001a37a00, 0x1dd18e0, 0xc0028805c0, 0x1df1458, 0xc0003bc770, 0x45998a, 0xc00004c000)
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20210502142502-4d8770ff3de8/vfilter.go:1345 +0x66
May 12 13:49:51 hostname velociraptor[461]: www.velocidex.com/golang/vfilter._MultiplicationExpression.Reduce(0xc001a37a00, 0x0, 0x0, 0x0, 0x1dd18e0, 0xc0028805c0, 0x1df1458, 0xc0003bc770, 0x8, 0x1a)
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20210502142502-4d8770ff3de8/vfilter.go:1634 +0x65
May 12 13:49:51 hostname velociraptor[461]: www.velocidex.com/golang/vfilter._AdditionExpression.Reduce(0xc001a37a20, 0x0, 0x0, 0x0, 0x1dd18e0, 0xc0028805c0, 0x1df1458, 0xc0003bc770, 0xc00006fbc8, 0x203000)
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20210502142502-4d8770ff3de8/vfilter.go:1525 +0x85
May 12 13:49:51 hostname velociraptor[461]: www.velocidex.com/golang/vfilter._ConditionOperand.Reduce(0x0, 0xc001a37a40, 0x0, 0x1dd18e0, 0xc0028805c0, 0x1df1458, 0xc0003bc770, 0x203000, 0x42467b)
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20210502142502-4d8770ff3de8/vfilter.go:1572 +0xac
May 12 13:49:51 hostname velociraptor[461]: www.velocidex.com/golang/vfilter._OrExpression.Reduce(0xc000406d08, 0x0, 0x0, 0x0, 0x1dd18e0, 0xc0028805c0, 0x1df1458, 0xc0003bc770, 0x203000, 0x203000)
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20210502142502-4d8770ff3de8/vfilter.go:1483 +0x78
May 12 13:49:51 hostname velociraptor[461]: www.velocidex.com/golang/vfilter._AndExpression.Reduce(0xc001a37a60, 0x0, 0x0, 0x0, 0x1dd18e0, 0xc0028805c0, 0x1df1458, 0xc0003bc770, 0xc002881540, 0x40)
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20210502142502-4d8770ff3de8/vfilter.go:1442 +0x85
May 12 13:49:51 hostname velociraptor[461]: www.velocidex.com/golang/vfilter.(*_AliasedExpression).Reduce(0xc002880d40, 0x1dd18e0, 0xc0028805c0, 0x1df1458, 0xc0003bc770, 0xc001152d00, 0x1b7aca0)
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20210502142502-4d8770ff3de8/vfilter.go:869 +0x41c
May 12 13:49:51 hostname velociraptor[461]: www.velocidex.com/golang/vfilter.(*_SelectExpression).Transform.func2(0x1dd18e0, 0xc0028805c0, 0x1df1458, 0xc0003bc690, 0xc001152d88, 0xc0049ede01)
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-202105021425024d8770ff3de8/vfilter.go:1115 +0x50
May 12 13:49:51 hostname velociraptor[461]: www.velocidex.com/golang/vfilter.MaterializedLazyRow(0x1dd18e0, 0xc0028805c0, 0x18e3440, 0xc0003bc700, 0x1df1458, 0xc0003bc690, 0xc0003bc620)
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20210502142502-4d8770ff3de8/lazy.go:96 +0x1f7
May 12 13:49:51 hostname velociraptor[461]: www.velocidex.com/golang/vfilter.(*_Select).processSingleRow(0xc002880d80, 0x1dd18e0, 0xc0028805c0, 0x1df1458, 0xc0003bc620, 0x1ad5a60, 0xc0003bc620, 0xc001b6ec60)
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20210502142502-4d8770ff3de8/vfilter.go:765 +0x5b4
May 12 13:49:51 hostname velociraptor[461]: www.velocidex.com/golang/vfilter.(*_Select).Eval.func4(0xc002880d80, 0x1dd18e0, 0xc0028805c0, 0x1df1458, 0xc0003bc620, 0xc001b6ec60)
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20210502142502-4d8770ff3de8/vfilter.go:744 +0x114
May 12 13:49:51 hostname velociraptor[461]: created by www.velocidex.com/golang/vfilter.(*_Select).Eval
May 12 13:49:51 hostname velociraptor[461]: #011/home/mic/go/pkg/mod/www.velocidex.com/golang/vfilter@v0.0.0-20210502142502-4d8770ff3de8/vfilter.go:729 +0x245
```

**Steps to reproduce (haven't confirmed, but should work):**
1. Collect `Windows.KapeFiles.Triage` using an offline collector
2. Unzip the file and remove the artifact files (leaving the `C` folder tree)
3. Re-zip the results
4. Attempt to import them with `System.Util.ImportCollection`
5. Observe the crash

**Expected behavior:** error rather than crash, ideally with a descriptive enough message to help diagnose ("no artifact json files found," or whatever).

I know this is surely an edge case, but any time an input can crash the program it seems high-priority. 
 Thanks!

**Comment by scudette** [comment]
Agree this should not crash given a malformed input :-). The zip files Velociraptor itself produces do not have directory entries (i.e. zip members representing directories) and the code did not expect those.  While a zip program would add those by default. 

Without the json files the import will not work anyway because we have no idea what artifact was actually imported (we determine the artifact name by the json file name).

**Comment by scudette** [comment]
This is now fixed in 0.6.0 RC1
